dbtwiz/generate/model.py

```diff
@@ -0,0 +1,156 @@
+import os
+import yaml
+
+from dbtwiz.interact import (
+    input_text,
+    select_from_list,
+    multiselect_from_list,
+    autocomplete_from_list,
+)
+
+from dbtwiz.logging import info, warn, error, fatal
+
+from dbtwiz.model import (
+    Group,
+    Project,
+    layer_choices,
+    materialization_choices,
+    access_choices,
+    domains_for_layer,
+    model_base_path,
+)
+
+
+def generate_model(quick: bool):
+    """Generate new dbt model"""
+
+    project = Project()
+
+    try:
+        layer = select_from_list(
+            "Select model layer",
+            layer_choices())
+
+        domain = autocomplete_from_list(
+            "Which domain does your model belong to",
+            domains_for_layer(layer),
+            must_exist=False,
+            allow_blank=False)
+
+        while True:
+            name = input_text(
+                "What is the name of your model")
+            base_path = model_base_path(layer, domain, name)
+            sql_path = base_path.with_suffix(".sql")
+            yml_path = base_path.with_suffix(".yml")
+            if sql_path.exists() or yml_path.exists():
+                error("A model with that name already exists, please choose another.")
+            else:
+                break
+
+        description = input_text(
+            "Give a short description of your model and its purpose")
+
+        if not quick:
+            group = autocomplete_from_list(
+                "Which group does your model belong to",
+                Group().choices(),
+                must_exist=True,
+                allow_blank=True)
+
+            access = select_from_list(
+                "What is the access level for this model",
+                access_choices())
+
+            materialization = select_from_list(
+                "How should your model be materialized",
+                materialization_choices())
+
+            teams = multiselect_from_list(
+                "Which team(s) should be responsible for this model",
+                project.teams())
+
+            service_consumers = access_policy = None
+
+            if layer in ("marts", "bespoke"):
+                service_consumers = multiselect_from_list(
+                    "Which service consumers need access to your model",
+                    project.service_consumers())
+
+                access_policy = select_from_list(
```

Comment:
It says `allow_none=True`, but I don't see how it should be possible to not select one of the options:
![image](https://github.com/user-attachments/assets/b45ab82f-87dd-4517-9851-e5d6152ffbdb)

Reply:
I now append `None` as the last choice when `allow_none=True`.